portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +28,13 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<String> messages = new ArrayList<String>();
+    messages.add("first message");
+    messages.add("second message");
+    messages.add("third message");
+    Gson gson = new Gson();
+    String jsonMessage = gson.toJson(messages);
```

Comment:
It's better to put the json conversion into a separate function because it is likely to be reused for most of your server responses.

Reply:
later will refactor, I think it's not good to do premature optimization(?)

Reply:
Sure, you can refactor later :)